pilot/pkg/xds/workloadentry.go

```diff
@@ -91,6 +95,14 @@ func (sg *InternalGen) RegisterWorkload(proxy *model.Proxy, con *Connection) err
 	entry := workloadEntryFromGroup(entryName, proxy, groupCfg)
 	setConnectMeta(entry, sg.Server.instanceID, con)
 	_, err = sg.store.Create(*entry)
+
+	if errors.IsAlreadyExists(err) {
+		// the initial Patch may have failed due to a fast-reconnect and slow propagation in the API server
```

Comment:
Including `NotFound` in the Patch `retry.OnError` would cause us to retry multiple times even during happy-path registration.